.github/scripts/validate-e2e-page-object-usage.ts

```diff
@@ -0,0 +1,131 @@
+            console.error('No artifacts found for changed files. Exiting with failure.');
+            process.exit(1);
+        }
+
+        // Parse the changed and new files with status
+        const changedAndNewFilePathsWithStatus = changedFilesContent.split('\n').filter(line => line.trim() !== '').map(line => {
+            const [status, filePath] = line.split('\t');
+            return { status, filePath };
+        });
+
+        // Filter files based on the provided fileStatus
+        const filteredFiles = changedAndNewFilePathsWithStatus.filter(file => {
+            if (fileStatus === 'both') {
+                return file.status === 'A' || file.status === 'M';
+            }
+            return file.status === fileStatus;
+        }).map(file => file.filePath);
+
+        e2eFiles = filterE2eChangedFiles(filteredFiles);
+        console.log('Filtered E2E files:', e2eFiles);
+    } else {
+        // Running locally
+        console.log('Running locally, performing git diff against main branch...');
+        const diffOutput = execSync('git diff --name-status main...HEAD').toString().trim();
+        const changedFiles = diffOutput.split('\n').filter(line => line.trim() !== '').map(line => {
+            const [status, filePath] = line.split('\t');
+            return { status, filePath };
+        });
+
+        // Filter files based on the provided fileStatus
+        const filteredFiles = changedFiles.filter(file => {
+            if (fileStatus === 'both') {
+                return file.status === 'A' || file.status === 'M';
+            }
+            return file.status === fileStatus;
+        }).map(file => file.filePath);
+
+        e2eFiles = filterE2eChangedFiles(filteredFiles);
+        console.log('Filtered E2E files:', e2eFiles);
+    }
+
+    if (e2eFiles.length === 0) {
+        console.log('No E2E files to validate. Exiting successfully.');
+        process.exit(0);
+    }
+
+    // Check each E2E file for page object usage
+    for (const file of e2eFiles) {
+        try {
+            const content = fs.readFileSync(file, 'utf8');
+            // Check for the presence of page object imports
+            const usesPageObjectModel = content.includes('./page-objects/');
+
+            if (!usesPageObjectModel) {
+                console.error(`\x1b[31mFailure: You need to use Page Object Model in ${file}\x1b[0m`);
+                process.exit(1);
+            }
+        } catch (error) {
+            if (error.code === 'ENOENT') {
+                console.warn(`File not found: ${file}`);
+                continue; // Skip this file because it was deleted, so no need to validate
+            } else {
+                throw error; // Re-throw if it's a different error
+            }
+        }
+    }
+
+    console.log("\x1b[32mSuccess: All the new or modified E2E files use the Page Object Model.\x1b[0m");
+}
+
+// Run the verification for new files only
+verifyE2ePageObjectsUsage('A').catch((error) => {
```

Comment:
in the future this can be changed, to 'both' whenever we want to verify both Added and Modified files